tests/test_stateful_browser.py

```diff
@@ -339,8 +339,10 @@ def make_file(content):
                     ("first file content", "second file content"))
 
     # The form doesn't have a type=file field, but the target action
-    # does show it => add the fields ourselves.
+    # does show it => add the fields ourselves, and add enctype too.
     browser.select_form()
+    browser._StatefulBrowser__state.form.form[
+      "enctype"] = "multipart/form-data"
```

Comment:
The form works within a browser, so it should work so you shouldn't need this.

Reply:
The httpbin form is not in multipart/form-data, so you won't be able to upload without adding it, even in a browser

Reply:
Oops, indeed. The form doesn't have a file entry, but this piece of code is about adding it, you're right.